### DISCUSSION TITLE
Add Bulk Update permission

### DISCUSSION THREAD

**Discussion by matthiashamacher** [pimcore/pimcore#10933] [Q&A / Support]
Hey all,

I want to add  a permission for bulk updates. The general implementation of the permission is complete and I also tested everything by changing the core code of pimcore. But now I want to implement it correctly by overriding the function for batch edit which is this one: https://github.com/pimcore/pimcore/blob/10.x/bundles/AdminBundle/Resources/public/js/pimcore/element/helpers/gridColumnConfig.js#L461-#L505

My problem is, that all my efforts to override the function do not work. I tried `Ext.override` in an extra file, in the pimcoreReady function in startup.js, registering the complete class again but nothing seem to work. The files are always loaded and console.logs before the override always execute, but nothing I add to the function itself is executed and printed to the console.

Therefore my question is: how can I override that function in Pimcore X?

**Comment by ibrushgit** [comment]
The thing is that helpers are not used directly, but added to classes similar to php traits. For instance, gridColumnConfig helper is appended to listfolder like that: 
`pimcore.asset.listfolder.addMethods(pimcore.element.helpers.gridColumnConfig);`

We run into the same issue and resolved it via creating a new helper class, and then adding its methods (with same names as original) to all affected classes like:
`pimcore.element.selector.object.addMethods(pimcore.element.helpers.gridColumnConfigExtended);
pimcore.object.search.addMethods(pimcore.element.helpers.gridColumnConfigExtended);
pimcore.object.variantsTab.addMethods(pimcore.element.helpers.gridColumnConfigExtended);`

hope it helps someone
